src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -1941,23 +1764,18 @@ public SolutionState WithFrozenPartialCompilationIncludingSpecificDocument(Docum
         /// <summary>
         /// Creates a new solution instance with all the documents specified updated to have the same specified text.
         /// </summary>
-        public SolutionState WithDocumentText(IEnumerable<DocumentId> documentIds, SourceText text, PreservationMode mode)
+        public SolutionState WithDocumentText(IEnumerable<DocumentId?> documentIds, SourceText text, PreservationMode mode)
         {
-            if (documentIds == null)
-            {
-                throw new ArgumentNullException(nameof(documentIds));
-            }
-
-            if (text == null)
-            {
-                throw new ArgumentNullException(nameof(text));
-            }
-
             var solution = this;
 
             foreach (var documentId in documentIds)
             {
-                var doc = solution.GetDocumentState(documentId);
+                if (documentId == null)
+                {
+                    continue;
+                }
```

Comment:
I get you're maintaining behavior....this is also terrifying. Any idea if this is a recent regression or has been around for forever?

Reply:
IDK, would need to dig into history.

Reply:
This method also skips document ids that are not in the solution. Is that terrifying as well? If not then null document id is also not in the solution, so at least that's consistent.

Reply:
Thanks for filing #42029 on this one.